hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntityDelayedFetchInitializer.java

```diff
@@ -121,16 +117,23 @@ public void resolveInstance(RowProcessingState rowProcessingState) {
 						final PersistenceContext persistenceContext = session.getPersistenceContextInternal();
 						entityInstance = persistenceContext.getEntity( euk );
 						if ( entityInstance == null ) {
-							entityInstance = ( (UniqueKeyLoadable) concreteDescriptor ).loadByUniqueKey(
-									uniqueKeyPropertyName,
-									identifier,
-									session
-							);
-
-							// If the entity was not in the Persistence Context, but was found now,
-							// add it to the Persistence Context
-							if ( entityInstance != null ) {
-								persistenceContext.addEntity( euk, entityInstance );
+							if ( ( (AbstractEntityInitializer) parentAccess ).getEntityDescriptor()
+									.getBytecodeEnhancementMetadata()
+									.isEnhancedForLazyLoading() ) {
+								entityInstance = persistenceContext.proxyFor( entityInstance );
+							}
```

Comment:
This doesn't make much sense as `entityInstance` is always null at this point. Maybe you left this around while trying out some stuff?

Reply:
right, my bad it should be
```
if ( ((AbstractEntityInitializer) this.parentAccess).getEntityDescriptor()
									.getBytecodeEnhancementMetadata()
									.isEnhancedForLazyLoading()  ) {
								return;
							}
```